src/Dfc.CourseDirectory.Database/Pttcd/StoredProcedures/RefreshFindACourseIndex.sql

```diff
@@ -0,0 +1,160 @@
+﻿CREATE PROCEDURE [Pttcd].[RefreshFindACourseIndex]
+	@CourseRunIds [Pttcd].[GuidIdTable] READONLY
+AS
+BEGIN
+
+	WITH RecordsCte AS (
+		SELECT
+			-- Id is the CourseRun's ID appended with the Region, if there is one
+			CONVERT(VARCHAR(36), cr.CourseRunId) + CASE WHEN crr.RegionId IS NOT NULL THEN '-' + crr.RegionId ELSE '' END AS Id,
+			1 AS OfferingType, -- Course
+			c.CourseId,
+			cr.CourseRunId,
+			c.UpdatedOn,
+			p.ProviderId,
+			p.ProviderName AS ProviderDisplayName,
+			c.ProviderUkprn,
+			ld.LearnAimRefTitle QualificationCourseTitle,
+			c.LearnAimRef,
+			ld.NotionalNVQLevelv2,
+			c.CourseDescription,
+			cr.CourseName,
+			cr.DeliveryMode,
+			cr.FlexibleStartDate,
+			cr.StartDate,
+			cr.Cost,
+			cr.CostDescription,
+			cr.DurationUnit,
+			cr.DurationValue,
+			cr.StudyMode,
+			CASE WHEN cr.DeliveryMode = 1 THEN cr.AttendancePattern ELSE null END AttendancePattern,
+			cr.[National],
+			v.VenueName,
+
+			CONCAT_WS(
+				', ',
+				NULLIF(v.AddressLine1, ''),
+				NULLIF(v.AddressLine2, ''),
+				NULLIF(v.Town, ''),
+				NULLIF(v.County, ''),
+				NULLIF(v.Postcode, '')
+			) AS VenueAddress,
+
+			v.Town AS VenueTown,
+			COALESCE(v.Position, r.Position) AS Position,
+			r.Name AS RegionName,
+
+			-- Magic numbers and logic from https://github.com/SkillsFundingAgency/dfc-providerportal-changefeedlistener/commit/608340dcfaa5c74ee8b1ae422ad902ee0c529c01#diff-5f9ef9c9ca0b0bc9af8b5c7926cfcfe31fa7e8367b9104357104ad568fbe0302R103-R104
+			CAST(CASE
+				WHEN r.RegionId IS NULL THEN 1
+				WHEN r.ParentRegionId IS NOT NULL THEN 4.5  -- Sub region
```

Comment:
You can name numbers in sql like this, a bit like a c# const.

```
declare @sub_region int = 4.5;
...
update x set y = @sub_region where z;
```
Don't worry about changing them now though, it's fine as is.

Reply:
This whole thing needs reviewing, I think. I'm pretty sure one of the three cases here can *never* happen but it was ported over directly from the CFL for a first pass.